validation-test/Sema/type_checker_crashers_fixed/rdar116122902.swift

```diff
@@ -0,0 +1,23 @@
+// RUN: %target-typecheck-verify-swift
+
+protocol AnyValue {
+}
+
+struct Value<T> {}
+
+extension Value: AnyValue where T = { // expected-error {{use '==' for same-type requirements rather than '='}} expected-error {{expected type}}
+// expected-note@-1 {{requirement from conditional conformance of 'Value<T>' to 'AnyValue'}}
+}
+
+struct Test {
+  var tuple: (value: any AnyValue, id: Int)?
+
+  mutating func test<T>(v: Value<T>) {
+    _ = {
+      self.tuple = (v, 42)
+      // expected-error@-1 {{cannot assign value of type '(Value<T>, Int)' to type '(value: any AnyValue, id: Int)'}}
+      // expected-error@-2 {{generic struct 'Value' requires the types 'T' and '<<error type>>' be equivalent}}
```

Comment:
Looks like this `<<error type>>` here is not intended
Maybe worth a `FIXME` here?

Reply:
No worries, I think it's useful to show the exact requirement that caused the failure even if one side of it is invalid and I don't know if it would be possible to avoid mentioning `<<error type>>`

Reply:
Yeah, maybe...